src/message.rs

```diff
@@ -269,7 +269,7 @@ pub struct Message {
     pub(crate) starred: bool,
     pub(crate) chat_blocked: Blocked,
     pub(crate) location_id: u32,
-    pub(crate) error: String,
+    pub(crate) error: Option<String>,
```

Comment:
Since you provide an accessor function as Rust public API, why not make this entirely private field so that the dc code also has to use the public API?  It seems it's only assigned to in this module here anyway.